pkg/apis/dmi/v1beta1/api.proto

```diff
@@ -0,0 +1,349 @@
+// to make the mapper maintain the list of device information.
+service DeviceMapperService {
+    // RegisterDevice registers a device to the device mapper.
+    // Device manager registers a device instance with the information of device
+    // to the mapper through the interface of RegisterDevice.
+    // When the mapper gets the request of register with device information,
+    // it should add the device to the device list and connect to the real physical device via the specific protocol.
+    rpc RegisterDevice(RegisterDeviceRequest) returns (RegisterDeviceResponse) {}
+    // RemoveDevice unregisters a device to the device mapper.
+    // Device manager unregisters a device instance with the name of device
+    // to the mapper through the interface of RemoveDevice.
+    // When the mapper gets the request of unregister with device name,
+    // it should remove the device from the device list and disconnect to the real physical device.
+    rpc RemoveDevice(RemoveDeviceRequest) returns (RemoveDeviceResponse) {}
+    // UpdateDevice updates a device to the device mapper
+    // Device manager updates the information of a device used by the mapper
+    // through the interface of UpdateDevice.
+    // The information of a device includes the meta data and the status data of a device.
+    // When the mapper gets the request of updating with the information of a device,
+    // it should update the device of the device list and connect to the real physical device via the updated information.
+    rpc UpdateDevice(UpdateDeviceRequest) returns (UpdateDeviceResponse) {}
+    // CreateDeviceModel creates a device model to the device mapper.
+    // Device manager sends the information of device model to the mapper
+    // through the interface of CreateDeviceModel.
+    // When the mapper gets the request of creating with the information of device model,
+    // it should create a new device model to the list of device models.
+    rpc CreateDeviceModel(CreateDeviceModelRequest) returns (CreateDeviceModelResponse) {}
+    // RemoveDeviceModel remove a device model to the device mapper.
+    // Device manager sends the name of device model to the mapper
+    // through the interface of RemoveDeviceModel.
+    // When the mapper gets the request of removing with the name of device model,
+    // it should remove the device model to the list of device models.
+    rpc RemoveDeviceModel(RemoveDeviceModelRequest) returns (RemoveDeviceModelResponse) {}
+    // UpdateDeviceModel update a device model to the device mapper.
+    // Device manager sends the information of device model to the mapper
+    // through the interface of UpdateDeviceModel.
+    // When the mapper gets the request of updating with the information of device model,
+    // it should update the device model to the list of device models.
+    rpc UpdateDeviceModel(UpdateDeviceModelRequest) returns (UpdateDeviceModelResponse) {}
```

Comment:
The DeviceModel only services the DeviceInstance. Do you add the DeviceModel API to the DMI to write all the logic in mapper? Otherwise, is the definition of DeviceModel API necessary?

Reply:
No, we need to send deviceModel to mapper, which will align better with the native design